### PR TITLE
[r] Fix missing organ value of CGM files (#2735)

### PR DESCRIPTION
**Pull request by danielsotirhos**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented on demo expectations                        <sub>or labelled PR as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that Demo expectations are clear              <sub>or PR is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and build passes in `sandbox` <sub>or added `no sandbox` label</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [ ] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [ ] Added PR reference to merge commit title
- [ ] Collected commit title tags in merge commit title
- [ ] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [ ] Purchased BQ slot committment in `dev`                <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Deleted BQ slot committment in `dev`                  <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Purchased BQ slot committment in `prod`               <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Deleted BQ slot committment in `prod`                 <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

File: src/azul/plugins/metadata/hca/transform.py [hunk 1]
```diff
@@ -1131,14 +1131,12 @@ def matrix_stratification_values(self, file: api.File) -> JSON:
                     contents['donors'].append(donors)
                 organ = stratum.get('organ')
                 if organ is not None:
-                    organ = sorted(organ)
-                    contents['samples'].append(
+                    organ = ', '.join(sorted(organ))
```
Comment (amarjandu) [inline src/azul/plugins/metadata/hca/transform.py:1134]: I assume the space after `,` is intentional?

**Review by amarjandu — APPROVED** [review]

**Comment by amarjandu** [comment]
@danielsotirhos i think you can retry the test and get the coverage results.

File: src/azul/plugins/metadata/hca/transform.py (hunk 1 above)
Comment (danielsotirhos) [reply src/azul/plugins/metadata/hca/transform.py:1134]: Thank you for pointing this out. Although the `specimens` inner entity field `organ` is string, joining the (possibly multiple) organ values provided by the CGM's metadata into one string is incorrect. Instead separate `specimens` inner entity should be added for each organ value.

File: src/azul/plugins/metadata/hca/transform.py [hunk 2]
```diff
@@ -1132,15 +1132,13 @@ def matrix_stratification_values(self, file: api.File) -> JSON:
...
-                            'biomaterial_id': f'specimen_from_organism_{file_name}',
-                            'entity_type': 'specimens',
-                            'organ': organ,
-                            'effective_organ': organ,
-                        }
-                    )
+                    for i, one_organ in enumerate(sorted(organ)):
+                        contents['specimens'].append(
+                            {
+                                'biomaterial_id': f'specimen_from_organism_{i}_{file_name}',
```
Comment (hannes-ucsc) [inline src/azul/plugins/metadata/hca/transform.py:1138]:
```suggestion
                                'biomaterial_id': f'specimen_from_organism_{i}_{file_name}',
```

How come this change isn't reflected in any of the tests? Should I at least see `_0` in some of the assertions?

File: test/service/test_response.py [hunk 3]
```diff
@@ -1940,7 +1940,7 @@ def test_contributor_matrix_files(self):
             ('developmentStage', 'adult'): [
                 '4d6f6c96-2a83-43d8-8fe1-0f53bffd4674.HumanLiver.zip',
             ],
-            ('organ', 'liver'): [
+            ('specimenOrgan', 'liver'): [
```
Comment (hannes-ucsc) [inline test/service/test_response.py:1943]: And what's this? This seems unrelated. Or is this the real fix and the enumeration above is cosmetic?

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Some issues here with how this is broken up into commits and/or test coverage. But happy to discuss. Maybe I'm misunderstanding something.

**Comment by hannes-ucsc** [comment]
Before I forget, @danielsotirhos. I'd rather not see fixups during my initial review.

**Comment by danielsotirhos** [comment]
Reviewer questions addressed in PL.

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
Before you merge this, @danielsotirhos, please add a screenshot to the description of #2735 so we have a before/after for the demo.

For demo, attempt to reproduce the issue.
